scripts/manage_pes

```diff
@@ -148,456 +89,355 @@ USAGE, OPTIONS, AND EXAMPLES
+
+Add Mode
+   manage_pes  --add --file <pes_filename> --pes_setby <component_name>
+"""
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+############################################################################
+def parse_command_line(args, description):
+############################################################################
+    parser = argparse.ArgumentParser(
+        usagetext,
+        description=description,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    CIME.utils.setup_standard_logging_options(parser)
+
+    parser.add_argument("-add", "--add", action="store_true", default=False,
+                        help="add a new pes layout")
+    parser.add_argument("-pes_setby", "--pes_setby",
+                        choices=['allactive', 'cam', 'cice', 'cism', 'clm',
+                                 'drv', 'pop'],
+                        help="component name")
+    parser.add_argument("-file", "--file", default=None,
+                        help="test list to add")
+    parser.add_argument("-grid", "--grid", default=None,
+                        help="a regular expression that matches the grid naming"
+                        "convention")
+    parser.add_argument("-query", "--query", action="store_true", default=False,
+                        help="Query the peslist by grid and machine")
+    parser.add_argument("-machine", "--machine", default=None,
+                        help="can be a supported machine name")
+    args = parser.parse_args(args[1:])
+    CIME.utils.handle_standard_logging_options(args)
+
+    return (args.file, args.add, args.query, args.pes_setby,
+            args.grid, args.machine)
+
+class Manage_Pes:
+    ADD_MODE = 1
+    QUERY_MODE = 2
+
+    def __init__(self, filename, add, query, setby, grid, machine):
+        self.filename = filename
+        self.add = add
+        self.query = query
+        self.model = CIME.utils.get_model()
+        self.setby = setby
+        self.grid = grid
+        self.machine = machine
+
+    def checkargs(self):
+        logger.debug("filename=%s", self.filename)
+        logger.debug("add=%s", self.add)
+        logger.debug("query=%s", self.query)
+        logger.debug("model=%s", self.model)
+        logger.debug("setby=%s", self.setby)
+        logger.debug("grid=%s", self.grid)
+        logger.debug("machine=%s", self.machine)
+
+        if self.add and self.query:
+            return False
+        if not self.add and not self.query:
+            return False
+
+        if self.add:
+            self.mode = Manage_Pes.ADD_MODE
+            if self.filename is None:
+                logger.warning("No filename given for add mode")
+                return False
+            if self.setby is None:
+                logger.warning("No pes_setby given for add mode")
+                return False
+        else:
+            self.mode = Manage_Pes.QUERY_MODE
+
+        return True
+
+    def usage(self):
```

Comment:
Agreed, please check argument sanity in parse_command_line. You can then just expect/assert that there's no contradictions in Manage_Pes.